Il2CppInterop.Runtime/InteropTypes/Arrays/Il2CppArrayBase.cs

```diff
@@ -18,6 +24,13 @@ private protected static bool ThrowImmutableLength()
     {
         throw new NotSupportedException("Arrays have immutable length");
     }
+
+    private protected void ThrowIfIndexOutOfRange(int index)
+    {
+        if (index < 0 || index >= Length)
```

Comment:
dotnet/runtime uses the `(uint)index >= (uint)Length` pattern a lot[^1], should we do so aswell?

[^1]: https://github.com/search?q=repo%3Adotnet%2Fruntime+%22%28uint%29index+%3E%3D+%28uint%29%22&type=code

Reply:
Implemented